packages/ai/src/ai/common/config.py

```
@@ -145,7 +145,14 @@ def merge(userConfig: Dict[str, Any], defaultConfig: Dict[str, Any]) -> Dict[str
                 raise Exception(f'Default profile {profile} is not defined in {logicalType}')
 
             # Get the settings for this default name
-            defaultConfig = preconfig['profiles'].get(profile)
+            profileConfig = preconfig['profiles'].get(profile)
+
+            # Check if default profile is deprecated
+            if isinstance(profileConfig, dict) and profileConfig.get('deprecated'):
+                migration_msg = profileConfig.get('migration', 'Please use a current profile instead.')
+                warning(f'Default profile "{profile}" is deprecated. {migration_msg}')
+
+            defaultConfig = profileConfig if isinstance(profileConfig, dict) else profileConfig
```

Ruff (0.15.7)

[warning] Useless `if`-`else` condition

(RUF034)